connectivity/check/deployment.go

```diff
@@ -451,9 +455,83 @@ func (ct *ConnectivityTest) validateDeployment(ctx context.Context) error {
 		}
 	}
 
+	// Set the timeout for all IP cache lookup retries
+	ipCacheCtx, cancel := context.WithTimeout(ctx, ct.params.ipCacheTimeout())
+	defer cancel()
+	for _, cp := range ct.ciliumPods {
+		err := ct.waitForIPCache(ipCacheCtx, cp)
+		if err != nil {
+			return err
+		}
+	}
+
 	return nil
 }
 
+func (ct *ConnectivityTest) waitForIPCache(ctx context.Context, pod Pod) error {
```

Comment:
It'll be good to keep some stats around latency/number of retries requires so that we can track any regressions in the code. Can be done in a follow-up.

Reply:
In this case it seems better to me to check ipcache state before running the tests. This is because all the test pod deployments are done before any tests are run, so ipcache changes after the initial flux are not to be expected. Will need to revisit if we change to add/delete pods between tests.